ign_rviz_plugins/src/rviz/plugins/GlobalOptions.cpp

```diff
@@ -31,6 +31,70 @@ namespace rviz
 {
 namespace plugins
 {
+////////////////////////////////////////////////////////////////////////////////
+TFStatus::TFStatus()
+: status(" "), message(" "), color("green"), currentState(0)
+{}
+
+////////////////////////////////////////////////////////////////////////////////
+bool TFStatus::update(std::string & _fixedFrame, std::vector<std::string> & _allFrames)
+{
+  // Check for tf data
+  auto framePosition = std::find(_allFrames.begin(), _allFrames.end(), _fixedFrame);
+
+  if (!_allFrames.size()) {
+    std::string msg = "No tf data. Frame [" + _fixedFrame + "] does not exist";
+
+    this->status = QString::fromStdString("Fixed Frame [Warn]");
+    this->message = QString::fromStdString(msg);
+    this->color = QString::fromStdString("orange");
+    if (currentState != 1) {
+      currentState = 1;
+      return true;
+    }
+  } else if (framePosition == _allFrames.end()) {
+    std::string msg = "Frame [" + _fixedFrame + "] does not exist";
+
+    this->status = QString::fromStdString("Fixed Frame [Error]");
+    this->message = QString::fromStdString(msg);
+    this->color = QString::fromStdString("red");
+
+    if (currentState != 2) {
+      currentState = 2;
+      return true;
+    }
```

Comment:
Are these checks really necessary?

Reply:
These checks are performed to check whether the status has been changed. We can remove them and call `emit tfStatusChanged()` on every render event.

Reply:
I think that will be better. Let's remove them for now. The updates on the render event will be helpful because:
1. The text `Frame[<fixedframe>] does not exist` text will be updated when the Frame is changed and the "currentState" is still the same. (In case of warning/error)
2. The implementation of the `TFStatus` will be modified in the future when the TF Status support for multiple frames is added.